### PR TITLE
remove joblib recipe

### PR DESCRIPTION
**Pull request by kain88-de**
The version here is 2 years old. The conda-forge version is better maintained. On conda-forge I can install 0.11. on this channel only version 0.9.3 (from 2015) is available.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by mbargull** [comment]
Please only remove the recipe itself. Don't change the dependencies of the other packages (those are minimum requirements anyways, hence the newer releases from `conda-forge` always satisfy them).

**Comment by kain88-de** [comment]
> (those are minimum requirements anyways, hence the newer releases from conda-forge always satisfy them).

That is the point. They should be forced to pick up the newer version from conda-forge. This is the only sane way I know of to remove a package with the channel pinning from conda forge. For example, if you have a condarc like this.

```yaml
channels:
  - bioconda
  - conda-forge
  - defaults
```

Then conda will *always* use the old joblib version available in bioconda when it's installed as a dependency. Requiring a larger version number will ensure that the newer package version is installed.

**Comment by mbargull** [comment]
Ah, I must suddenly became colorblind as I thought you removed the constraints rather than add them. Really sorry about that!
Yes, without those minimum version requirements this is currently something you can only make `conda` do with `--no-channel-priority`. So I understand the need for this workaround but I am not happy with it -- and cannot offer an alternative at the same time. I'd actually like to here some other opinions on this matter. @bioconda/core: thoughts?
If we'll add those constraints, we should add at least a comment as to why this was done.

**Comment by bgruening** [comment]
@mbargull thats the only solution that works at the moment. An other way, also suboptimal, is to mark the bioconda package as `broken` and remove the `main` label.

**Comment by kain88-de** [comment]
> If we'll add those constraints, we should add at least a comment as to why this was done.

I can add a comment once the discussion has reached a conclusion.

**Comment by mbargull** [comment]
@bgruening: Yeah, it's a bit like choosing between pest and cholera. Seeing that there is only `joblib 0.9.3` and not multiple versions on the Bioconda channel, I'd lean more to making them unavailable via a label. The existing label `deprecated` would be more appropriate than `broken`, though.

> I can add a comment once the discussion has reached a conclusion.

@kain88-de: Thank you for that and thanks for doing this overall!

**Comment by kain88-de** [comment]
> @kain88-de: Thank you for that and thanks for doing this overall!

This version problem has cost me 20min of work time today. Fixing it took less time (yay emacs) and hopefully helps others avoid this.

**Comment by bgruening** [comment]
Ok, I will deprecate it. @kain88-de can you revert the strict pinning?

**Comment by kain88-de** [comment]
done

File: recipes/triform2/meta.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ requirements:
     - pandas
     - numpy
     - natsort
-    - joblib
+    - jobli
```
Comment (bgruening) [inline recipes/triform2/meta.yaml]: this looks wrong

File: recipes/pomegranate/meta.yaml [hunk 2]
```diff
@@ -15,15 +15,15 @@ requirements:
     - python
     - cython >=0.22.1
     - numpy >=1.8.0
-    - joblib >=0.9.0b4
+    - joblib
```
Comment (bgruening) [inline recipes/pomegranate/meta.yaml]: Not sure why this restriction was in place, but better leave it.

**Comment by kain88-de** [comment]
Oh yeah sorry. You can change it yourself. Github allows you to push to my repo on that branch. 

 I won’t get back to this before Monday.

**Pull request merged**